### PR TITLE
Add description table for install files for Knative Serving

### LINKED ISSUE #3274: (Split of #1896) Add technical reference for install files for Knative Serving
**Describe the change you'd like to see**
This issues is a child issue to #1896. The only thing required for the "closing" of this issue is to create a markdown table of all of the Assets listed in the Serving release. 

In Scope: 
- list of installation files
- table the describes the file and identifies dependencies/requirements

STORY 
```
AS A New User

I WANT to understand what each of the Assets in a Knative Serving Release is for 

SO THAT if I am preforming a custom install, I have reference documentation
```

Install files for Serving can be found under "Assets" here: https://github.com/knative/serving/releases/tag/v0.21.0

NOTE: unclear where this content goes as of yet, but don't let that stop you from making progress! It's just a 3 column .MD table after all :)

### PR DESCRIPTION
**Pull request by yanweiguo**
Fixes #3274.

### CODE REVIEW

**Comment by yanweiguo** [comment]
/assign @ZhiminXiang @dprotaso @ricardozanini 

cc @omerbensaadon

**Comment by omerbensaadon** [comment]
thank you @yanweiguo!! 

@snneji @RichardJJG can one of you give this a review?

**Comment by RichardJJG** [comment]
@omerbensaadon, yep, will do.

**Review by RichardJJG — APPROVED** [review]
Needs a small change or two, which I'll take care of.

File: docs/install/assets/serving.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Knative Serving installation files (0.21 releases)
+
+| File name | Description | Dependencies|
+| --- | --- | --- |
+| serving-core.yaml | Knative Serving core components. | serving-crds.yaml |
+| serving-crds.yaml | Knative Serving core CRDs. | none |
+| serving-default-domain.yaml | Configuring Knative Serving to use <a href="http://xip.io">xip.io</a> as the default DNS suffix. | serving-core.yaml |
+| serving-domainmapping-crds.yaml | CRDs required by the DomainMapping feature. | none |
```
Comment (RichardJJG) [inline docs/install/assets/serving.md:8]: | serving-domainmapping-crds.yaml | CRDs required by the Domain Mapping feature. | none |

File: docs/install/assets/serving.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+| serving-domainmapping.yaml | Componnents required by the DomainMapping feature. | serving-domainmapping-crds.yaml |
```
Comment (RichardJJG) [inline docs/install/assets/serving.md:9]: | serving-domainmapping.yaml | Components required by the Domain Mapping feature. | serving-domainmapping-crds.yaml |

File: docs/install/assets/serving.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+| serving-hpa.yaml | Components to auto scale Knative revisions via Kubeneters HPA. | serving-core.yaml |
```
Comment (RichardJJG) [inline docs/install/assets/serving.md:10]: | serving-hpa.yaml | Components to autoscale Knative revisions through the Kubernetes Horizontal Pod Autoscaler. | serving-core.yaml |

File: docs/install/assets/serving.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  serving-nscert.yaml | Components to provision TLS wildcard certificates. | serving-core.yaml |
+| serving-post-install-jobs.yaml | Optional jobs after installing serving-core.yaml. Currently it is same to `serving-storage-version-migration.yaml`. | serving-core.yaml |
```
Comment (RichardJJG) [inline docs/install/assets/serving.md:12]: | serving-post-install-jobs.yaml | Optional jobs after installing serving-core.yaml. Currently it is the same as `serving-storage-version-migration.yaml`. | serving-core.yaml |

File: docs/install/assets/serving.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+| serving-storage-version-migration.yaml | Migrating the storage version of Knative resources, including Service, Route, Revision, Configuration, from v1alpha1 & v1beta1 to v1. Required by upgrade from 0.18 to 0.19 version. | serving-core.yaml |
```
Comment (RichardJJG) [inline docs/install/assets/serving.md:13]: | serving-storage-version-migration.yaml | Migrates the storage version of Knative resources, including Service, Route, Revision, and Configuration, from `v1alpha1` and `v1beta1` to `v1`. Required by upgrade from version 0.18 to 0.19. | serving-core.yaml |

**Review by RichardJJG — CHANGES_REQUESTED** [review]
I'm new to GitHub reviews; ignore my last "approve" review. This one has all the changes I wanted to make.

**Comment by omerbensaadon** [comment]
@markusthoemmes @davetropeano @yanweiguo see above changes by Richard! Thank you all :)

**Comment by yanweiguo** [comment]
@RichardJJG PTAL

**Comment by snneji** [comment]
Whoops added the wrong label! I wanted to test using the Prow bot.

**Comment by dprotaso** [comment]
looks good if we need further edits we can make a new PR

/lgtm
/approve
/hold cancel

**Pull request merged**